### DISCUSSION TITLE
Built-in virtual hosts ?

### DISCUSSION THREAD

**Discussion by i-Ching** [litestar-org/litestar#939] [Q&A] [answered]
Does Starlite support **built-in** multiple hosting ?

I have tested another frameworks:
1. Starlette: Yes, its implemenation is the most concise so far: See Host-based routing of https://www.starlette.io/routing/

2. Sanic: Yes, just add @app.route("/", host="sub.example.com:8000")
See https://github.com/sanic-org/sanic/blob/main/examples/vhosts.py

3. Quart: Yes, the concept is same as Flask. But I need to make many steps to make it work
Reference: https://ron.sh/two-domains-one-flask/

4. aiohttp: No, but its official document has just passed it to Nginx or Apache, blah blah blah ....
See: https://docs.aiohttp.org/en/stable/deployment.html
In other words, aiohttp does NOT have this function.

5. Before I make a comparision report of above frameworks, I want to make sure if Starlite provide built-in function of virtual hosting?

**Comment by provinzkraut** [comment]
This is not supported and support for this is not planned. It's recommended to defer this part of the routing logic to a proxy server.

**Comment by i-Ching** [comment, marked as answer]
provinzkraut,

Thank you for your clear answer. Finally I get a clear marking scheme at built-in virtual hosts feature:

No: **aiohttp**
Yes: Quart
Yes: Sanic
Yes: Starlette
No: **Starlite**

You are better than the official documentation of [aiohttp](https://docs.aiohttp.org/en/stable/deployment.html).
aiohttp wastes my reading on its rhetoric paragraph to defer this feature to NGINX or Apache in order to avoid the absence of function.
